pkg/fleet-manager/clusters.go

```diff
@@ -36,20 +40,47 @@ import (
 	fleetapi "kurator.dev/kurator/pkg/apis/fleet/v1alpha1"
 )
 
+type ClusterInterface interface {
+	IsReady() bool
+	GetObject() client.Object
+	GetSecretName() string
+	GetSecretKey() string
+}
+
+const (
+	ClusterKind         = "Cluster"
+	AttachedClusterKind = "AttachedCluster"
+)
+
 func (f *FleetManager) reconcileClusters(ctx context.Context, fleet *fleetapi.Fleet) (ctrl.Result, error) {
 	fleetKey := client.ObjectKeyFromObject(fleet)
 	log := ctrl.LoggerFrom(ctx).WithValues("fleet", fleetKey)
 	var unreadyClusters int32
 	var result ctrl.Result
-	var readyClusters []clusterv1alpha1.Cluster
+	var readyClusters []ClusterInterface
 	clusterMap := make(map[string]struct{}, len(fleet.Spec.Clusters))
 	// Loop over cluster, and add labels to the cluster
 	for _, cluster := range fleet.Spec.Clusters {
-		clusterMap[cluster.Name] = struct{}{}
 		// cluster namespace can be not set, always use fleet namespace as a fleet can only include clusters in the same namespace.
 		clusterKey := types.NamespacedName{Name: cluster.Name, Namespace: fleet.Namespace}
-		var cluster clusterv1alpha1.Cluster
-		err := f.Get(ctx, clusterKey, &cluster)
+		var currentCLuster ClusterInterface
+		var err error
+		if cluster.Kind == ClusterKind {
+			var cluster clusterv1alpha1.Cluster
+			err = f.Get(ctx, clusterKey, &cluster)
+			currentCLuster = &cluster
+		} else if cluster.Kind == AttachedClusterKind {
```

Comment:
nit: We can also support custom cluster, not a must now

Reply:
will add this in a subsequent pr